Fixes #23275: move subscriptions page to main app.

waldenraines opened the pull request:
Replace the existing angular subscriptions page with a new react
version of the page.

http://projects.theforeman.org/issues/23275

bbuckingham commented:
Before this is merged, please make sure that the new subscriptions page is functional on nightly (via Lab Features). (It may already be; however, I have not confirmed.)

waldenraines commented:
> Before this is merged, please make sure that the new subscriptions page is functional on nightly (via Lab Features). (It may already be; however, I have not confirmed.)

Will do, thanks for the reminder.

amirfefer commented on engines/bastion_katello/app/assets/javascripts/bastion_katello/subscriptions/details/views/subscription-details.html:
```diff
@@ -1,25 +0,0 @@
-<div data-extend-template="layouts/details-page-with-breadcrumbs.html">
```
a subscription details page presented as a blank page: 
![empty_subscription](https://user-images.githubusercontent.com/11807069/40385729-3ef3b92a-5e10-11e8-9a1a-ccae8b3342dc.png)

looks like there's no react details page

waldenraines replied:
See http://projects.theforeman.org/issues/23687

amirfefer commented on webpack/containers/Application/config.js:
```diff
@@ -11,11 +11,11 @@ export const links = [
   {
     text: 'RH Subscriptions',
-    path: 'xui/subscriptions',
+    path: 'subscriptions',
```
the subscriptions page renders with empty state, even when I have subscriptions on the old page:
![screenshot from 2018-05-22 16-42-25](https://user-images.githubusercontent.com/11807069/40386196-b4bf6f04-5e11-11e8-953e-f42310017d1e.png)


The problem here is probably because somehow katello uses react 16.2, as you can see here:


![screenshot from 2018-05-22 22-27-45](https://user-images.githubusercontent.com/11807069/40385924-e48df9cc-5e10-11e8-9fe4-a93124754ef2.png)

but we use `static getDerivedStateFromProps` in `SubscriptionTable.js` which is defined on 16.3. switching this to `componentWillReceiveProps` works well (although this lifecycle method is deprecated).

nitpick: could you place this spinner to the center of the page:
![screenshot from 2018-05-22 22-27-46](https://user-images.githubusercontent.com/11807069/40386353-2a9e0dac-5e12-11e8-9360-9ccce47b9eef.png)

waldenraines replied:
> The problem here is probably because somehow katello uses react 16.2, as you can see here:

See http://projects.theforeman.org/issues/23690 and http://projects.theforeman.org/issues/23691

> nitpick: would this spinner look better on the center of the page ?

@rohoover I will change this to whatever you think is best, just let me know.

Rohoover replied:
@waldenraines 
Thanks for bringing this to my attention.

Yes please center it. Here you can see some of the design guidelines (http://www.patternfly.org/pattern-library/communication/loading-state/#design). It doesn't seem to mention the space at the top, so if you double the current space at the top that would be good with me.

waldenraines replied:
> Yes please center it.

Since the spinner itself isn't really part of this work I created a new issue for it - http://projects.theforeman.org/issues/23922

waldenraines replied:
And PR: https://github.com/Katello/katello/pull/7451

johnpmitsch commented:
@waldenraines can you remove the experimental UI setting?

johnpmitsch commented:
@waldenraines is this still valid? https://github.com/Katello/katello/blob/master/webpack/index.js#L13-L16

waldenraines commented:
> @waldenraines can you remove the experimental UI setting?

If you mean the Labs menu menu item, no, because it will be used in the future.

johnpmitsch commented:
@waldenraines looks like this is still going to `xui`? https://git.io/f4bBT

johnpmitsch commented:
@waldenraines I'm confused, you want to remove the experimental UI [here](https://git.io/f4bB3), but keep the setting? What does the setting currently do?

waldenraines commented:
> What does the setting currently do?

Sorry, should have explained this better.  The setting creates a menu called "Experimental Labs" and allows additions to that menu.  For all I know, other plugins may have already added things to the menu and the setting isn't even defined in katello, it's defined in foreman, so I won't be removing it in this PR.  Moreover, I'm 95% sure that the experimental labs menu will be used for other things in the near future (such as host UI merge).

> @waldenraines I'm confused, you want to remove the experimental UI here, but keep the setting? 

I removed the experimental UI subscriptions menu item yes.  If you wanted to add something to that menu in the future you could add something similar to what I just added.  The menu itself is still there it's just not being shown because nothing is added to it.  

> @waldenraines is this still valid? https://github.com/Katello/katello/blob/master/webpack/index.js#L13-L16

Yeah, but the name should be changed to something other than xui.  Updated.

> @waldenraines looks like this is still going to xui? https://git.io/f4bBT

Ah, blast.  I must have forgotten to search again for `xui` after reworking this PR, updated.  Good catch.

johnpmitsch commented:
@waldenraines thanks for the explanation. I think my confusion was that the menu doesn't show up (screenshot is after a page refresh). I'm guessing it won't show up if its empty. Since this is used across plugins though, there isn't much we can do about that, even if it means having a setting that "does nothing" in this case.
![here](http://www.clipular.com/c/6736158072242176.png?k=Trx6VPcSvatlNA2hjbb3cN1BFEQ)
